### DISCUSSION TITLE
Screen Flashing

### DISCUSSION THREAD

**Discussion by eecue** [Jyers/Marlin#365] [Q&A] [answered]
Hello,

I flashed the screen, at first it wasn't working but then I reformatted as FAT-32 with 4k allocation and that worked.  The blue stayed on for a few seconds then went orange.  However I do not see the custom icons.  Did I miss a step?

**Comment by Jyers** [comment, marked as answer]
To get the custom icons you need to compile your own firmware with CREALITY_DWIN_EXTUI_CUSTOM_ICONS enabled.

**Reply by eecue** [reply]
Oops, I just took the folder that was in there!

**Comment by eecue** [comment]
Ok I just built the firmware using your config and I uncommented CREALITY_DWIN_EXTUI_CUSTOM_ICONS:

```
//
// Ender-3 v2 OEM display. A DWIN display with Rotary Encoder.
//
#define CREALITY_DWIN_EXTUI
#if ENABLED(CREALITY_DWIN_EXTUI)
  //
  // Enable custom icons
  // NB: Requires Ender-3 v2 OEM display firmware update, or you will get blank icons!
  //
  #define CREALITY_DWIN_EXTUI_CUSTOM_ICONS
#endif

```

The build works fine but I'm only seeing the main firmware, not the display firmware:


```
ls -l .pio/build/STM32F103RET6_creality
total 55608
drwxr-xr-x@ 61 eecue  staff      1952 Apr  1 19:39 FrameworkArduino
drwxr-xr-x@  5 eecue  staff       160 Apr  1 19:38 FrameworkArduinoVariant
-rwxr-xr-x   1 eecue  staff    256360 Apr  1 19:39 firmware-20210401-193833.bin
-rwxr-xr-x   1 eecue  staff  13971544 Apr  1 19:39 firmware-20210401-193833.elf
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 lib155
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 lib2b5
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 lib38f
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 lib448
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 lib7ea
drwxr-xr-x@  4 eecue  staff       128 Apr  1 19:39 libc03
drwxr-xr-x@  3 eecue  staff        96 Apr  1 19:39 src
```

**Comment by eecue** [comment]
Oh I get it now, I _did_ flash my display correctly, now I just need to flash the firmware I just built! 

Edit: that worked! I see cute mesh icons now!

**Reply by Jyers** [reply]
Sorry if it was a bit confusing, I really need to make a good guide for the whole screen flashing, but I haven't even had time to do it myself... Anyway, glad you got it working!

**Reply by eecue** [reply]
Here's the steps I followed if you want to use them for docs:

1) Format a microSD card (must be 32GB or smaller, I used the 8GB one that came with the printer).  Make sure you format FAT32 with 4k allocation (which was the default allocation for me using OS X Disk Utility)
2) Download the full project https://github.com/Jyers/Marlin and then copy the complete DWIN_SET folder (https://github.com/Jyers/Marlin/tree/Ender_3_V2_Extensible_UI/Display%20firmware/DWIN_SET) as is to the root of your SD card so that you see a folder called DWIN_SET on the SD card
3) Power off the printer and take off the display, pull out the ribbon cable and remove the 4 hex screws from the back of the display.  Use the silver studs as a lever to open the display, prying them to the right when you have the ribbon connector port to the top.  This will release the back cover.
4) Put the SD card into the slot and plug back in the ribbon connector.
5) Power up the printer, the screen should be a solid blue for 3-5 seconds and then turn and orange/red.  If it flashes blue and goes to orange/red you need to double check you formatted the card properly.  Put the screen case back together and mount back on the printer.
6) Copy the two config files from https://github.com/Jyers/Marlin/tree/Ender_3_V2_Extensible_UI/Configuration%20Files/Jyers's%20Config into the Marlin subfolder (not the top level project file): https://github.com/Jyers/Marlin/tree/Ender_3_V2_Extensible_UI/Marlin
7) (I found this guide helpful: https://automatedhome.party/2020/05/08/compiling-marlin-firmware-is-easy-dont-be-afraid-to-do-it/) Install Python and VSCode if you didn't already have them 
8) Install the Auto Build Marlin plugin https://marketplace.visualstudio.com/items?itemName=MarlinFirmware.auto-build
9) Edit the config files you copied and set anything you want differently, I personally changed some temperature settings
10) Uncomment the line by removing the `//` in front of `#define CREALITY_DWIN_EXTUI_CUSTOM_ICONS` so it looks like: 
```
//
// Ender-3 v2 OEM display. A DWIN display with Rotary Encoder.
//
#define CREALITY_DWIN_EXTUI
#if ENABLED(CREALITY_DWIN_EXTUI)
  //
  // Enable custom icons
  // NB: Requires Ender-3 v2 OEM display firmware update, or you will get blank icons!
  //
  #define CREALITY_DWIN_EXTUI_CUSTOM_ICONS
#endif
```
11) Build the firmware by clicking build in the Marlin build panel
12) Open the build output after the build, it's in the root of the Marlin project in a folder called `.pio/build/STM32F103RET6_creality/` and copy the .bin file to a blank formatted SD card as above
13) Put the SD card in your printer's main slot and power it up.  You should see a grey/dark blue screen for a minute and then the boot screen.  Go into the Leveling screen and you should see cute mesh icons.
